Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -415,4 +423,37 @@ theorem length_reverse {a : FreeMonoid α} : a.reverse.length = a.length :=
 
 end Reverse
 
+section IsomorphicTypes
+
+variable {α β : Type*}
+
+/-- free monoids over isomorphic types are isomorphic -/
+@[to_additive "if two types are isomorphic, the additive free monoids over those types are
+isomorphic"]
+def congrEquiv (e : α ≃ β) : FreeMonoid α ≃* FreeMonoid β :=
```

Comment:
Free groups call it `FreeGroup.freeGroupCongr`, quotient groups use `QuotientGroup.congr`. Basic types use `MulEquiv.prodCongr` etc. I would stick to one of the existing naming schemes (also, we should unify them some day).

Reply:
done!